fuseimg/data/ops/shape_ops.py

```diff
@@ -82,7 +81,214 @@ def __call__(self, sample_dict: NDict, op_id: Optional[str], key: str,
+                                                           outer_width=resize_to[1] + 2 * padding[1], pad_value=self.pad_value)
+        return image
+        
+    def pad_inner_image(self,image: np.ndarray, outer_height: int, outer_width: int, pad_value: float) -> np.ndarray :
+        """
+        Pastes input image in the middle of a larger one
+        :param image:        image of shape [H, W, C]
+        :param outer_height: final outer height
+        :param outer_width:  final outer width
+        :param pad_value:    value for padding around inner image
+        :number_of_channels  final number of channels in the image
+        :return:             padded image
+        """
+        inner_height, inner_width = image.shape[0], image.shape[1]
+        h_offset = int((outer_height - inner_height) / 2.0)
+        w_offset = int((outer_width - inner_width) / 2.0)
+        if self.number_of_channels > 1 :
+            outer_image = np.ones((outer_height, outer_width, self.number_of_channels), dtype=image.dtype) * pad_value
+            outer_image[h_offset:h_offset + inner_height, w_offset:w_offset + inner_width, :] = image
+        elif self.number_of_channels == 1 :
+            outer_image = np.ones((outer_height, outer_width), dtype=image.dtype) * pad_value
+            outer_image[h_offset:h_offset + inner_height, w_offset:w_offset + inner_width] = image
+        return outer_image    
+    
+    def resize_image(self,inner_image: np.ndarray, resize_to: Tuple[int,int]) -> np.ndarray :
+        """
+        resize image to the required resolution
+        :param inner_image: image of shape [H, W, C]
+        :param resize_to: required resolution [height, width]
+        :return: resized image
+        """
+        inner_image_height, inner_image_width = inner_image.shape[0], inner_image.shape[1]
+        if inner_image_height > resize_to[0]:
+            h_ratio = resize_to[0] / inner_image_height
+        else:
+            h_ratio = 1
+        if inner_image_width > resize_to[1]:
+            w_ratio = resize_to[1] / inner_image_width
+        else:
+            w_ratio = 1
+    
+        resize_ratio = min(h_ratio, w_ratio)
+        if resize_ratio != 1:
+            inner_image = skimage.transform.resize(inner_image,
+                                                   output_shape=(int(inner_image_height * resize_ratio),
+                                                                 int(inner_image_width * resize_ratio)),
+                                                   mode='reflect',
+                                                   anti_aliasing=True
+                                                   )
+        return inner_image
+    
+    def __call__(self, sample_dict: NDict, key: str , resize_to: Tuple,
+                 padding: Tuple
+        ):
+        '''
+        :param resize_to:               new size of input images, keeping proportions
+        :param padding:                 padding size
```

Comment:
What is the format of padding -  there are few possible options.

Reply:
do you mean how much you pad in each size? 
right now we have 2 values for each axis, and it is symmetric on each size

Reply:
it's here in the documentation also
        pads image to requested size ,
        pads both side equally by the same input padding size (left = right = padding[1] , up = down= padding[0] )  ,
        padding default value is zero or minimum value in normalized target range